### ISSUE TITLE
How can I use array variable in check assertion? - try variable without ``

### ISSUE THREAD

**Issue by gauravgandhi15** [kelaberetiv/TagUI#726]
I have to trying to use array variable for check "contains":

`check filtersText contains `temp` | 'search box is visible - ' filtersText | 'search box is not visible - ' filtersText
`
temp is my array variable:

```
cardArt = ['', '14', '3', '6', '2', '4', '5', '3', '8']'
temp = cardArt[1]
```

In this scenario, it always accept "temp" as string and not as variable.

can you please help with this? I have tried to follow your documentation to use variable as `variable` but it doesn't work.

**Comment by kensoh** [comment]
Hi @gauravgandhi15 this is an interesting use case. Try the following to use temp without backticks. For check step, variables don't need backticks.

```
check filtersText contains temp | 'search box is visible - ' filtersText | 'search box is not visible - ' filtersText
```

 I know this is confusing, the original design was to simplify the syntax for certain steps such as echo / dump / write / check. But in doing so, it is inconsistent with the rest of the TagUI steps. In an upcoming v6 release we plan to fix this to make every step consistent to use backticks when a variable is required and no need for quotation marks for text strings.

**Comment by gauravgandhi15** [comment]
@kensoh i have tried without “ or ‘. Then it complaints about missing quotes.

**Comment by kensoh** [comment]
Oh sorry I got it, this is an edge case that is not supported natively. This is because TagUI tries to parse the natural language into JavaScript condition and would need either the single or double quote in order to know how to do that translation.

TagUI support conditions such as `if a contains '1' and a not contains '2'`, so the quotation marks have to be explicitly provided and not inferred if it is a variable. If we change the logic to let TagUI guess when it is a variable, above multi condition I think will fail.

Long story short, you can try below and it should work for this edge case. It works by explicitly providing the quotation mark but ending it right away so that context switches to variable, before enclosing in another dummy quotation marks pair.

```
filtersText = 'A14B'
cardArt = ['', '14', '3', '6', '2', '4', '5', '3', '8']
temp = cardArt[1]
check filtersText contains ''+temp+'' | 'search box is visible - ' filtersText | 'search box is not visible - ' filtersText
```

**Comment by kensoh** [comment]
Closing issue for now until further inputs, it should work. Let me know if there is issue running it!

**Comment by kensoh** [comment]
CC @siowyisheng

**Comment by gauravgandhi15** [comment]
Hey it works. Thank you so much for your help. I really appreciate

**Comment by kensoh** [comment]
Thanks @gauravgandhi15 for confirming! And sorry for the lack of support for this use case.

There were various considerations to weigh when I was designing this step. So I favoured the ability to write complex multiple conditions over the ability to support comparing a condition with a text string that is stored in a variable.

**Comment by GarneshMike** [comment]
Hi Sifus,

My attempt to search a KEYWORD from a STRING using the IF ELSE function always return False if the KEYWORD is a variable (EXAMPLE 1).
But if I hardcode the KEYWORD, it works (EXAMPLE 2)

Anything that i missed.


// Tagui Block Starts - EXAMPLE 1

string = "this is a test"
filter = 'test'

echo String = `string`
echo Filter = `filter`

if string contain "`filter`"
	echo True
else
	echo False

// Tagui Block Ends - EXAMPLE1



// Tagui Block Starts - EXAMPLE2
string = "this is a test"
filter = 'test'

echo String = `string`
echo Filter = `filter`

if string contain "test1"
	echo True
else
	echo False

// Tagui Block Ends - EXAMPLE2

**Comment by kensoh** [comment]
I think that is a limitation due to its human language syntax. Maybe you can try using JavaScript directly, something like:

```
a = 'red tree'
b = 'tree'
if a.indexOf(b) > -1
    echo True
else
    echo False
```

Copying @ruthtoh who is maintaining TagUI project now..
